nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name(name)
+    name[0...10].capitalize
+  end
```

Comment:
- [ ] According to the project requirements, kindly make the correct_name method in the Nameable class raise a `NotimplementedError`. The method does not need to receive any argument, it should only raise the error.

- [ ] As demanded by the project requirements, kindly create a Decorator class that inherits the Nameable class. In the constructor assign a nameable object from params to an instance variable. Implement the correct_name method that returns the result of the correct_name method of the @nameable. The correct name method in the Decorator class should look like
```ruby
def correct_name
      @nameable.correct_name
    end
```

- [ ] Kindly add a CapitalizeDecorator class that inherits the Decorator class as demanded by project requirements. You should also implement a method correct_name that capitalizes the output of @nameable.correct_name.

- [ ] Kindly add a TrimmerDecorator class that inherits the Decorator class as demanded by project requirements. Also, implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.